library/std/src/os/wasi/io/fd.rs

```diff
@@ -0,0 +1,280 @@
+//! Owned and borrowed file descriptors.
```

Comment:
There seem to be two copies of some of the core types and traits, one for WASI and one for UNIX. I'd like to only have *one* copy of the types and traits, and only have bits in WASI and UNIX that are unique to those target families.

If need be, we could have an `io_safety_common` or something. But ideally, couldn't this all live under UNIX?

Reply:
I've now factored this out into an `fd` module that lives in library/std/src/os. A plausible future for it could include factoring out the `AsRawFd`/`IntoRawFd`/`FromRawFd` code as well.

I avoided the word "common" since it it's still platform-specific to platforms that use file descriptors, but I'm not attached to this.